Source/WebKit/NetworkProcess/SharedWorker/WebSharedWorkerServerToContextConnection.cpp

```diff
@@ -142,8 +142,8 @@ void WebSharedWorkerServerToContextConnection::launchSharedWorker(WebSharedWorke
         }
     }
     send(Messages::WebSharedWorkerContextManagerConnection::LaunchSharedWorker { sharedWorker.origin(), sharedWorker.identifier(), sharedWorker.workerOptions(), sharedWorker.fetchResult(), initializationData });
-    sharedWorker.forEachSharedWorkerObject([&](auto, auto& port) {
-        postConnectEvent(sharedWorker, port, [](bool) { });
+    sharedWorker.forEachSharedWorkerObject([connection = Ref { *this }, sharedWorker = Ref { sharedWorker }](auto, auto& port) {
+        connection->postConnectEvent(sharedWorker, port, [](bool) { });
```

Comment:
Ref'ing `this` here is causing a crash with `ASSERTION FAILED: !m_adoptionIsRequired` (linked below). 

This is because in `WebSharedWorkerServerToContextConnection`'s constructor, we pass `*this` to `WebSharedWorkerServer::addContextConnection` and that un-adopted reference is eventually used by `WebSharedWorkerServer` to call this function here--resulting in this crash. 

How can I fix this? It seems wrong to use adoptRef in the constructor: `server.addContextConnection(adoptRef(*this));`

https://ews-build.s3-us-west-2.amazonaws.com/macOS-Sequoia-Debug-WK2-Tests-EWS/9d666430-6824/imported/w3c/web-platform-tests/content-security-policy/sandbox/shared-worker-sandbox-crash-log.txt

Reply:
Call `relaxAdoptionRequirement();` in the class constructor.